bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagMainJournal.kt

```diff
@@ -0,0 +1,175 @@
+
+        // Polling interval in milliseconds for the high water check thread.
+        private const val highWaterPollingIntervalMS = 200L
+
+        /**
+         * The global instance of the main journal.
+         * You must call startInstance() before accessing this.
+         */
+        val instance: BugsnagMainJournal
+            get() = mainJournal ?: throw IllegalStateException("Must call startInstance() before accessing instance")
+
+        /**
+         * Start the main journal at the specified base path.
+         * Any existing journal at this location will be overwritten.
+         *
+         * NOTE: This must be called once (and only once) before accessing instance.
+         *
+         * @param logger A logger
+         * @param baseDocumentPath Path to a filename that will be used as the basis for all journal and snapshot files
+         * @param initialDocument The initial contents of the document. This will be deep copied.
+         * @return The contents of the previous journal (if any)
+         */
+        fun startInstance(
+            logger: Logger,
+            baseDocumentPath: File,
+            initialDocument: Map<String, Any>
+        ): MutableMap<in String, out Any>? {
+            synchronized(hasInitialized) {
+                if (hasInitialized) {
+                    return null
+                }
+
+                val previousContents = try {
+                    loadPrevious(baseDocumentPath)
+                } catch (exc: IOException) {
+                    null
+                }
+
+                val journal = BugsnagMainJournal(logger, baseDocumentPath, initialDocument)
+                mainJournal = journal
+                hasInitialized = true
+                journal.beginSnapshotHousekeepingThread()
+                return previousContents
+            }
+        }
+
+        internal var hasInitialized = false
+        internal var mainJournal: BugsnagMainJournal? = null
+
+        // Internal to keep it accessible to unit tests
+        internal fun loadPrevious(baseDocumentPath: File): MutableMap<in String, out Any> {
+            return JournaledDocument.loadDocumentContents(baseDocumentPath)
+        }
+
+        // Internal to keep it accessible to unit tests
+        internal fun initialDocumentContents(document: Map<String, Any>): Map<String, Any> {
+            val docWithVersionInfo = document.toMutableMap()
+            docWithVersionInfo[versionInfoKey] = mapOf(
+                journalTypeKey to journalType,
+                journalVersionKey to journalVersion
+            )
+            return docWithVersionInfo
+        }
+
+        /**
+         * Test interface to addCommand that removes all restrictions.
+         * This is required so that the unit tests can replace the entire document.
+         */
+        internal fun testintfAddCommand(documentPath: String, value: Any?) {
+            instance.journal.addCommand(documentPath, value)
+        }
```

Comment:
Given that all the variables here are `internal` is there any reason why these test-helper functions shouldn't be part of the Unit Test source set?